reconcilers/reconcilers_validate_test.go

```diff
@@ -500,6 +500,168 @@ func TestChildReconciler_validate(t *testing.T) {
 	}
 }
 
+func TestChildSetReconciler_validate(t *testing.T) {
```

Comment:
Could you add a test case where `DesiredChildren` returns error and make sure the reconciliation `shouldErr`?

Reply:
done